packaging/metadata.py

```diff
@@ -54,77 +51,112 @@ class DynamicField(enum.Enum):
 
 
 class Metadata:
+    """A class representing the `Core Metadata`_ for a project.
 
-    """
-    A representation of core metadata.
+    Every potential metadata field except for ``Metadata-Version`` is represented by a
+    parameter to the class' constructor. The required metadata can be passed in
+    positionally or via keyword, while all optional metadata can only be passed in via
+    keyword.
+
+    Every parameter has a matching attribute on instances, except for *name* (see
+    :attr:`display_name` and :attr:`canonical_name`). Any parameter that accepts an
+    :class:`~collections.abc.Iterable` is represented as a :class:`list` on the
+    corresponding attribute.
     """
 
     # A property named `display_name` exposes the value.
     _display_name: str
     # A property named `canonical_name` exposes the value.
-    _canonical_name: utils.NormalizedName
-    version: packaging_version.Version
-    platforms: list[str]
+    _canonical_name: NormalizedName
+    version: Version
+    platforms: List[str]
```

Comment:
How did you get pre-commit to not change that thanks to pyupgrade running?

Reply:
I'm guessing it's the removal of `from future import __annotations__`.